pallets/xcm-helper/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+							if let Some(asset) = Self::assets_mapping(withdrawal.asset_id) {
+								let multi_asset = MultiAsset {
+									id: asset,
+									fun: Fungibility::Fungible(withdrawal.amount),
+								};
+								if orml_xtokens::module::Pallet::<T>::transfer_multiassets(
+									RawOrigin::Signed(
+										T::AssetHandlerPalletId::get().into_account_truncating(),
+									)
+									.into(),
+									Box::new(multi_asset.into()),
+									0,
+									Box::new(destination.clone()),
+									WeightLimit::Unlimited,
+								)
+								.is_err()
+								{
+									failed_withdrawal.push(withdrawal.clone())
+								}
+							} else {
+								failed_withdrawal.push(withdrawal)
+							}
+						} else if Self::handle_deposit(withdrawal.clone(), destination).is_err() {
+							failed_withdrawal.push(withdrawal)
+						}
+					} else {
+						failed_withdrawal.push(withdrawal)
+					}
+				}
+			});
+			<FailedWithdrawals<T>>::insert(n, failed_withdrawal);
+			Weight::default()
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Whitelists Token .
+		///
+		/// # Parameters
+		///
+		/// * `token`: Token to be whitelisted.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::whitelist_token(1))]
+		pub fn whitelist_token(origin: OriginFor<T>, token: u128) -> DispatchResult {
+			T::AssetCreateUpdateOrigin::ensure_origin(origin)?;
+			let mut whitelisted_tokens = <WhitelistedTokens<T>>::get();
+			ensure!(!whitelisted_tokens.contains(&token), Error::<T>::TokenIsAlreadyWhitelisted);
+			whitelisted_tokens.push(token);
+			<WhitelistedTokens<T>>::put(whitelisted_tokens);
+			Self::deposit_event(Event::<T>::TokenWhitelistedForXcm(token));
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::transfer_fee(1))]
+		pub fn transfer_fee(origin: OriginFor<T>, to: T::AccountId) -> DispatchResult {
+			T::AssetCreateUpdateOrigin::ensure_origin(origin)?;
+			let from = T::AssetHandlerPalletId::get().into_account_truncating();
+			let amount = T::AssetManager::reducible_balance(T::NativeAssetId::get(), &from, true);
+			T::AssetManager::transfer(T::NativeAssetId::get(), &from, &to, amount, true)?;
+			Self::deposit_event(Event::<T>::XcmFeeTransferred(to, amount));
+			Ok(())
+		}
+
+		// TODO: This should be removed after testing before creating a release
+		#[pallet::call_index(4)]
+		#[pallet::weight(Weight::from_ref_time(10_000) + T::DbWeight::get().writes(1))]
+		pub fn mock_deposit(origin: OriginFor<T>, recipient: T::AccountId) -> DispatchResult {
+			ensure_signed(origin)?;
+			let asset = AssetId::Concrete(MultiLocation { parents: 1, interior: Junctions::Here });
+			let asset_id = Self::generate_asset_id_for_parachain(asset);
+			let deposit = Deposit {
+				recipient,
+				asset_id,
+				amount: 1_000_000_000_000_000u128,
+				extra: Vec::new(),
+			};
+			let network = T::ParachainNetworkId::get();
+			T::Executor::execute_withdrawals(network, deposit.encode()).unwrap();
+			Ok(())
+		}
```

Comment:
I assume this supposed to be removed before releasing

Reply:
Yeah .. I will remove this

Reply:
Removed in pr #47